File: 2132.Convert 1D Array Into 2D Array/solution.py
class Solution(object):
    def construct2DArray(self, original, n, m):
        """
        :type original: List[int]
        :type m: int
        :type n: int
        :rtype: List[List[int]]
        """
        res = [[0 for j in range(m)] for i in range(n)]
        if(len(original)>n*m or n*m>len(original)):
            return []
        counter = 0
        for i in range(n):
            for j in range(m):
                res[i][j] = original[counter]
                counter+=1

        return res
        

File: 2132.Convert 1D Array Into 2D Array/solution.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int n, int m) {
        vector<vector<int>>res(n,(vector<int> (m,0)));
        int size = original.size();
        if(size<n*m || size>n*m){
            // res.empty();
            vector<vector<int>> c;
            return c;
        }

        int counter = 0;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                res[i][j] = original[counter];
                counter++;
            }
        }
        return res;
    }
};